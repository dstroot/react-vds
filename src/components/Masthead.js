import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from 'logo.png';
import Particles from 'react-particles-js';
import ErrorBoundary from 'ErrorBoundary';
import * as routes from '../constants/routes';

// import 'animate.css' components
import 'animate.css/source/_base.css';
import 'animate.css/source/fading_entrances/fadeInDown.css';

const Masthead = () => {
  return (
    <main className="bd-masthead" id="content" role="main">
      <div className="container">
        <div className="row">
          <div className="col-5 mx-auto col-md-5 order-md-2 d-none d-md-block">
            <ErrorBoundary>
              <Particles
                params={{
                  particles: {
                    number: {
                      value: 55,
                      density: {
                        enable: true,
                        value_area: 500,
                      },
                    },
                    color: {
                      value: '#7256ad',
                    },
                    shape: {
                      type: 'circle',
                      stroke: {
                        width: 1,
                        color: '#7256ad',
                      },
                      polygon: {
                        nb_sides: 5,
                      },
                    },
                    opacity: {
                      value: 0.9,
                      random: false,
                      anim: {
                        enable: false,
                        speed: 0.5,
                        opacity_min: 0.1,
                        sync: false,
                      },
                    },
                    size: {
                      value: 3,
                      random: true,
                      anim: {
                        enable: false,
                        speed: 30,
                        size_min: 0.1,
                        sync: false,
                      },
                    },
                    line_linked: {
                      enable: true,
                      distance: 110,
                      color: '#7256ad',
                      opacity: 0.9,
                      width: 1,
                    },
                    move: {
                      enable: true,
                      speed: 2,
                      direction: 'none',
                      random: true,
                      straight: false,
                      out_mode: 'out',
                      bounce: false,
                      attract: {
                        enable: false,
                        rotateX: 600,
                        rotateY: 1200,
                      },
                    },
                  },
                  retina_detect: true,
                }}
                className={'canvas-container'}
              />
            </ErrorBoundary>
          </div>
          <div className="col-md-7 order-md-1 text-center text-md-left pr-md-5">
            <h1 className="display-4 mb-3 animated fadeInDown text-primary">
              <img
                className="d-inline-block align-top"
                src={logo}
                width="50"
                height="50"
                alt=""
              />{' '}
              Verite Data Science
            </h1>
            <p className="lead text-primary">INNOVATION, POWERED BY DATA</p>
            <p className="lead">
              Our clients demand deep industry knowledge and experience, coupled
              with innovation and the ability to roll up our sleeves and make
              business transformation happen. As technology rapidly evolves,
              it’s more important than ever to have an innovation advisor on
              your side. Whether it’s time to create a new vision or put one
              into place, we can help.
            </p>
            <div className="d-flex flex-column flex-md-row lead mb-3">
              <NavLink
                to={routes.ABOUT}
                className="btn btn-lg btn-outline-primary mt-3 mb-2 mb-md-0 mr-md-2"
              >
                About Verite Data Science
              </NavLink>
            </div>
          </div>
        </div>
      </div>
    </main>
  );
};

export default Masthead;
