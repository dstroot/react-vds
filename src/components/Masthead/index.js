import React from 'react';
import { Link } from '@reach/router';
import logo from 'media/logo.png';
import ParticlesPanel from 'components/ParticlesPanel';
import * as routes from 'constants/routes';

// import 'animate.css' components
import 'animate.css/source/_base.css';
import 'animate.css/source/fading_entrances/fadeInDown.css';

const Masthead = () => {
  return (
    <div className="container mt-5">
      <div className="row">
        <div className="col-5 mx-auto col-md-5 order-md-2 d-none d-md-block">
          <ParticlesPanel />
        </div>
        <div className="col-md-7 order-md-1 text-center text-md-left pr-md-5">
          <h1 className="display-4 mb-3 animated fadeInDown text-primary">
            <img
              className="d-inline-block align-top"
              src={logo}
              width="50"
              height="50"
              alt=""
            />{' '}
            Verite Data Science
          </h1>
          <p className="lead text-primary">INNOVATION, POWERED BY DATA</p>
          <p className="lead">
            Our clients demand deep industry knowledge and experience, coupled
            with innovation and the ability to roll up our sleeves and make
            business transformation happen. As technology rapidly evolves, it’s
            more important than ever to have an innovation advisor on your side.
            Whether it’s time to create a new vision or put one into place, we
            can help.
          </p>
          <div className="d-flex flex-column flex-md-row lead mb-3">
            <Link
              to={routes.ABOUT}
              className="btn btn-lg btn-outline-primary mt-3 mb-2 mb-md-0 mr-md-2"
            >
              About Verite Data Science
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Masthead;
