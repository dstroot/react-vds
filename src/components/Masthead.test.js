import React from 'react';
import Masthead from './Masthead';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

test('Masthead renders a masthead', () => {
  const component = renderer.create(
    <Router>
      <Masthead />
    </Router>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
