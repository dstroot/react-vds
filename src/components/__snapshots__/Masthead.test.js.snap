// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Masthead renders a masthead 1`] = `
<main
  className="bd-masthead"
  id="content"
  role="main"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-5 mx-auto col-md-5 order-md-2 d-none d-md-block"
      >
        <div
          className="canvas-container"
        >
          <canvas
            className={undefined}
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
      <div
        className="col-md-7 order-md-1 text-center text-md-left pr-md-5"
      >
        <h1
          className="display-4 mb-3 animated fadeInDown text-primary"
        >
          <img
            alt=""
            className="d-inline-block align-top"
            height="50"
            src="logo.png"
            width="50"
          />
           
          Verite Data Science
        </h1>
        <p
          className="lead text-primary"
        >
          INNOVATION, POWERED BY DATA
        </p>
        <p
          className="lead"
        >
          Our clients demand deep industry knowledge and experience, coupled with innovation and the ability to roll up our sleeves and make business transformation happen. As technology rapidly evolves, it’s more important than ever to have an innovation advisor on your side. Whether it’s time to create a new vision or put one into place, we can help.
        </p>
        <div
          className="d-flex flex-column flex-md-row lead mb-3"
        >
          <a
            aria-current={null}
            className="btn btn-lg btn-outline-primary mb-2 mb-md-0 mr-md-2"
            href="/about"
            onClick={[Function]}
            style={undefined}
          >
            About Verite Data Science
          </a>
        </div>
      </div>
    </div>
  </div>
</main>
`;
